isisd/isisd.c

```diff
@@ -2156,36 +2301,52 @@ static void common_isis_summary(struct vty *vty, struct isis *isis)
 	}
 }
 
+static void common_isis_summary(struct vty *vty, struct json_object *json,
+				struct isis *isis)
+{
+	if (json) {
+		common_isis_summary_json(json, isis);
+	} else {
+		common_isis_summary_vty(vty, isis);
+	}
+}
+
 DEFUN(show_isis_summary, show_isis_summary_cmd,
-      "show " PROTO_NAME " [vrf <NAME|all>] summary",
+      "show " PROTO_NAME " [vrf <NAME|all>] summary [json]",
       SHOW_STR PROTO_HELP VRF_CMD_HELP_STR
       "All VRFs\n"
+       "json output\n"
       "summary\n")
 {
 	struct listnode *node;
 	int idx_vrf = 0;
 	struct isis *isis;
 	const char *vrf_name = VRF_DEFAULT_NAME;
 	bool all_vrf = false;
+	bool uj = use_json(argc, argv);
+	json_object *json = NULL;
 
 	ISIS_FIND_VRF_ARGS(argv, argc, idx_vrf, vrf_name, all_vrf)
 	if (!im) {
 		vty_out(vty, PROTO_NAME " is not running\n");
 		return CMD_SUCCESS;
 	}
+	if (uj)
+		json = json_object_new_object();
 	if (vrf_name) {
 		if (all_vrf) {
 			for (ALL_LIST_ELEMENTS_RO(im->isis, node, isis))
-				common_isis_summary(vty, isis);
+				common_isis_summary(vty, json, isis);
 
 			return CMD_SUCCESS;
 		}
 		isis = isis_lookup_by_vrfname(vrf_name);
 		if (isis != NULL)
-			common_isis_summary(vty, isis);
+			common_isis_summary(vty, json, isis);
 	}
 
-	vty_out(vty, "\n");
```

Comment:
can you explain the vty_json command here.    I was looking at other json code and it looks like they adds a "}\n" to end of the output.

by removing this from the vty_out will it cause any issues for existing shows?

Reply:
I've copy this way of doing json/vty cmds from other examples (like in `isis_te.c`) and vty cmds works as expected.